### PR TITLE
refactor: add intelligent anomaly detection capability to the alarm

### LINKED ISSUE #404: Add intelligent anomaly detection capability to the alarm
### Refactor Request

The alarm module needs to add intelligent anomaly detection capability, integrate with anomaly detection algorithms, and increase anomaly detection trigger options.

### PR DESCRIPTION
**Pull request by wangsiyuan-code**
# Which issue does this PR close?

Closes #404 

# Rationale for this change
 
add intelligent anomaly detection capability to the alarm

# What changes are included in this PR?
Add the algorithm interface model in
 /holoinsight/server/home/home-alert/src/main/java/io/holoinsight/server/home/alert/model/compute/algorithm/anomaly

Added abnormal down trigger in
/holoinsight/server/home/home-alert/src/main/java/io/holoinsight/server/home/alert/service/calculate/AnomalyDownAbnormalDetect.java

Added abnormal up trigger in
/holoinsight/server/home/home-alert/src/main/java/io/holoinsight/server/home/alert/service/calculate/AnomalyUpAbnormalDetect.java

# Are there any user-facing changes?

none

# How does this change test

none

### CODE REVIEW

**Comment by masaimu** [comment]
You should fix "Merge Gate / merge-gate (pull_request_target)" first.

File: server/home/home-alert/src/main/java/io/holoinsight/server/home/alert/service/calculate/AnomalyDownAbnormalDetect.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    algorithmRequest.setIntervalTime(60000);
+    algorithmRequest.setDetectTime(functionConfigAIParam.getPeriod());
+    algorithmRequest.setAlgorithmConfig(algorithmConfig);
+    algorithmRequest.setRuleConfig(ruleConfig);
+    Map<String, Double> inputTimeSeries = new HashMap<>();
+    dataResult.getPoints().forEach((k, v) -> inputTimeSeries.put(k.toString(), v));
+    algorithmRequest.setInputTimeSeries(inputTimeSeries);
+
+    TriggerAIResult triggerAIResult = new TriggerAIResult();
+    // 设置算法接口名称
```
Comment (masaimu) [inline server/home/home-alert/src/main/java/io/holoinsight/server/home/alert/service/calculate/AnomalyDownAbnormalDetect.java:62]: Alter English comment.

File: server/home/home-alert/src/main/java/io/holoinsight/server/home/alert/service/calculate/AnomalyDownAbnormalDetect.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    String algoUrl = url + "/anomaly_detect";
+    // 调用算法接口
```
Comment (masaimu) [inline server/home/home-alert/src/main/java/io/holoinsight/server/home/alert/service/calculate/AnomalyDownAbnormalDetect.java:64]: Alter English comment.

File: server/home/home-alert/src/main/java/io/holoinsight/server/home/alert/service/calculate/AnomalyUpAbnormalDetect.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    algorithmRequest.setAlgorithmConfig(algorithmConfig);
+    algorithmRequest.setRuleConfig(ruleConfig);
+    Map<String, Double> inputTimeSeries = new HashMap<>();
+    dataResult.getPoints().forEach((k, v) -> inputTimeSeries.put(k.toString(), v));
+    algorithmRequest.setInputTimeSeries(inputTimeSeries);
+
+    TriggerAIResult triggerAIResult = new TriggerAIResult();
+    // 设置算法接口名称
+    String algoUrl = url + "/anomaly_detect";
+    // 调用算法接口
```
Comment (masaimu) [inline server/home/home-alert/src/main/java/io/holoinsight/server/home/alert/service/calculate/AnomalyUpAbnormalDetect.java:64]: Alter English comment.

File: server/home/home-alert/src/main/java/io/holoinsight/server/home/alert/service/calculate/AnomalyDownAbnormalDetect.java (hunk 1 above)
Comment (wangsiyuan-code) [reply server/home/home-alert/src/main/java/io/holoinsight/server/home/alert/service/calculate/AnomalyDownAbnormalDetect.java]: fixed

**Review by masaimu — APPROVED** [review]
LGTM

**Pull request merged**